### ISSUE TITLE
Detecting unused closure use variables

### ISSUE THREAD

**Issue by VasekPurchart** [slevomat/coding-standard#386]
Inspired by the recent unused/useless sniffs you were creating I think detecting unused uses in closures would be also helpful:

```
$foo = 'foo';
function () use ($foo) {
    return true;
}
```

Can't think of any false positive cases either right now.

Not sure about the name, but maybe something like `Functions.UnusedClosureUseVariable`?

**Comment by kukulich** [comment]
Something like https://github.com/slevomat/coding-standard/commit/171cb510f4f3b894557ff1940d4115cb3750ca25 ? :)

Still not sure about the name :)
